packages/story-editor/src/app/media/utils/useProcessMedia.js

```diff
@@ -478,11 +479,95 @@ function useProcessMedia({
     ]
   );
 
+  /**
+   * Crop video using FFmpeg.
+   *
+   * @param {import('@googleforcreators/media').Resource} resource Resource object.
+   * @param {Object<{newWidth: number, newHeight: number, cropElement: Element}>} cropParams Crop params.
+   */
+  const cropExistingVideo = useCallback(
+    ({ id: elementId, resource: oldResource }, cropParams) => {
+      const { id: resourceId, src: url, mimeType, isOptimized } = oldResource;
+      const { newWidth, newHeight, cropElement } = cropParams;
+
+      const onUploadSuccess = ({ id, resource }) => {
+        copyResourceData({ oldResource, resource });
+        updateOldTranscodedObject(oldResource.id, resource.id, 'source-video');
+        updateElementById({
+          elementId,
+          properties: {
+            x: cropElement.x < 0 ? 0 : cropElement.x,
+            y:
+              cropElement.y < DANGER_ZONE_HEIGHT
+                ? -DANGER_ZONE_HEIGHT
+                : cropElement.y,
+            width: newWidth,
+            height: newHeight,
+            resource: {
+              ...resource,
+              id: oldResource.id,
+              poster: null,
+              posterId: 0,
+            },
```

Comment:
this is incorrect. Updating the resource in the element here is not going to retrigger the poster generation.